presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java

```diff
@@ -391,15 +394,39 @@ public static Boolean notEqual(@SqlType(JSON) Slice leftJson, @SqlType(JSON) Sli
     }
 
     @ScalarOperator(IS_DISTINCT_FROM)
-    @SqlType(BOOLEAN)
-    public static boolean isDistinctFrom(@SqlType(JSON) Slice leftJson, @IsNull boolean leftNull, @SqlType(JSON) Slice rightJson, @IsNull boolean rightNull)
+    public static class JsonIsDistinctFrom
     {
-        if (leftNull != rightNull) {
-            return true;
+        @SqlType(BOOLEAN)
+        public static boolean isDistinctFrom(@SqlType(JSON) Slice leftJson, @IsNull boolean leftNull, @SqlType(JSON) Slice rightJson, @IsNull boolean rightNull)
+        {
+            if (leftNull != rightNull) {
+                return true;
+            }
+            if (leftNull) {
+                return false;
+            }
+            return notEqual(leftJson, rightJson);
         }
-        if (leftNull) {
-            return false;
+
+        @SqlType(BOOLEAN)
+        public static boolean isDistinctFrom(
+                @BlockPosition @SqlType(value = JSON, nativeContainerType = Slice.class) Block left,
```

Comment:
@findepi : This parameter is `Block`, and next parameter is required to be annotated with `BlockIndex`. This convention is  already used in aggregation framework.

Reply:
`@Block` wouldn't work, annotation classes and regular classes are in the same namespace.

The annotation can be dropped all together, just like how `@IsNull` is implemented. Parsing code would be a little more complicated that way, but not at all infeasible. The parsing would require lookahead because the type being `Block` isn't sufficient indication. But we already do look ahead for `@IsNull`.

I agree that `@BlockPosition` is a terrible name. I don't particularly like `@ContainingBlock` either.

Reply:
> `@Block` wouldn't work, annotation classes and regular classes are in the same namespace.

of course! I was only trying to convey my thinking what the ideal naming could be.

> Parsing code would be a little more complicated that way, but not at all infeasible. 

Surely my intention isn't to make things more complicated. Maybe we can think of some better name. Or maybe dropping the annotation is the way.

> For this PR, I'd leave this as is. This PR isn't introducing the syntax. It's just using it.

 sure!